src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala

```diff
@@ -0,0 +1,248 @@
+
+              Position
+        1 2 3 4 5 6 7 8 9 10
+  Gene1 ---
+  Gene2   -
+  Gene3 -----
+  */
+
+  val vdsBurden: VariantDataset = hc.importVCF("src/test/resources/regressionLinear.vcf")
+    .annotateVariantsIntervals("src/test/resources/regressionLinear.interval_list", "va.genes", all=true)
+    .annotateVariantsExpr("va.weight = v.start.toDouble")
+    .annotateSamplesTable("src/test/resources/regressionLinear.pheno",
+      "Sample",
+      root = Some("sa.pheno"),
+      config = TextTableConfiguration(types = Map("Pheno" -> TDouble), missing = "0"))
+    .annotateSamplesTable("src/test/resources/regressionLinear.cov",
+      "Sample",
+      root = Some("sa.cov"),
+      config = TextTableConfiguration(types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)))
+
+  def makeMap(sampleKT: KeyTable): Map[Any, IndexedSeq[java.lang.Double]] = sampleKT.collect().map { r =>
+    val s = r.asInstanceOf[Row].toSeq
+    s.head -> s.tail.map(_.asInstanceOf[java.lang.Double]).toIndexedSeq }
+    .toMap
+
+  def assertMap(mapR: Map[Int, IndexedSeq[java.lang.Double]], map: Map[Any, IndexedSeq[java.lang.Double]], tol: Double) {
+    (1 to 3).foreach { i =>
+      (mapR(i), map(s"Gene$i")).zipped.foreach( (x, y) =>
+        assert((x == null && y == null) || D_==(x.doubleValue(), y.doubleValue(), tolerance = tol))
+      )
+    }
+  }
+
+  @Test def testWeightedSum() {
+    val (linregKT, sampleKT) = vdsBurden.linregBurden("gene", "va.genes",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1", "sa.cov.Cov2"))
+
+    val linregMap = makeMap(linregKT)
+    val sampleMap = makeMap(sampleKT)
+
+    /*
+    Variant A       B       C       D       E       F       Weight
+    1       0.0     1.0     0.0     0.0     0.0     1.0     1.0
+    2         .     2.0       .     2.0     0.0     0.0     2.0
+    3       0.0       .     1.0     1.0     1.0       .     3.0
+
+    Sample  Cov1    Cov2    Pheno   Gene1   Gene2   Gene3
+    A       0.0     -1.0    1.0     0.0     0.0     0.0
+    B       2.0     3.0     1.0     5.0     4.0     5.0
+    C       1.0     5.0     2.0     0.0     0.0     3.0
+    D       -2.0    0.0     2.0     4.0     4.0     7.0
+    E       -2.0    -4.0    2.0     0.0     0.0     3.0
+    F       4.0     3.0     2.0     1.0     0.0     1.0
+    */
+
+    // Values from R: fit <- lm(Pheno ~ Gene1 + Cov1 + Cov2, data=df)
+    val linregMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(-0.08164, 0.15339, -0.532, 0.6478),
+      2 -> IndexedSeq(-0.09900, 0.17211, -0.575, 0.6233),
+      3 -> IndexedSeq(0.01558, 0.18323, 0.085, 0.940))
+
+    val sampleMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(0.0, 5.0, 0.0, 4.0, 0.0, 1.0),
+      2 -> IndexedSeq(0.0, 4.0, 0.0, 4.0, 0.0, 0.0),
+      3 -> IndexedSeq(0.0, 5.0, 3.0, 7.0, 3.0, 1.0))
+
+    assertMap(linregMapR, linregMap, 1e-3)
+    assertMap(sampleMapR, sampleMap, 1e-6)
```

Comment:
I think we should use [`mapSameElements`](https://github.com/hail-is/hail/blob/master/src/main/scala/is/hail/utils/package.scala#L310) for this. It compares two maps for equality and prints useful error messages when the key sets are non-equal, and when values are different, it prints useful context (i.e. the key). It takes a third argument for how to compare values, which AFAICT, should just be:

```scala
(x, y) => (x == null && y == null) || (x != null && y != null && D_==(x.doubleValue(), y.doubleValue(), tolerance = tol))
```

You'll have to preprocess one of your maps to get the key types equal, but it should just be `linregMapR.mapKeys(x => s"Gene$x")`